### PR TITLE
adding new constants and resources project as dependency and refactoring the code.

### PR DESCRIPTION
**Pull request by Aravinda93**
This project was previously using constant information from its own project. Now that we have the central openepcis-epcis-constants and openepcis-epcis-resources projects, we have added the dependency to this and used variables from there. Kindly request you to review and approve the PR.

### CODE REVIEW

File: src/main/java/io/openepcis/convert/VersionTransformer.java [hunk 1]
```diff
@@ -149,45 +152,49 @@ public final InputStream convert(
   public final InputStream convert(
       final InputStream inputDocument,
       final String fromMediaType,
```
Comment (sboeckelmann) [inline src/main/java/io/openepcis/convert/VersionTransformer.java:154]: also make use of new EPCISFormat

File: src/main/java/io/openepcis/convert/VersionTransformer.java [hunk 2]
```diff
@@ -149,45 +152,49 @@ public final InputStream convert(
...
-      final EpcisVersion fromVersion,
+      final EPCISVersion fromVersion,
       final String toMediaType,
```
Comment (sboeckelmann) [inline src/main/java/io/openepcis/convert/VersionTransformer.java:156]: also make use of new EPCISFormat

File: src/main/java/io/openepcis/convert/VersionTransformer.java [hunk 3]
```diff
@@ -149,45 +152,49 @@ public final InputStream convert(
...
       return convert(
           toXml(inputDocument),
-          "application/xml",
-          EpcisVersion.VERSION_2_0,
-          EpcisVersion.VERSION_1_2);
-    } else if (fromMediaType.toLowerCase().contains("xml")
-        && toMediaType.toLowerCase().contains("json")
-        && fromVersion.equals(EpcisVersion.VERSION_2_0)
-        && toVersion.equals(EpcisVersion.VERSION_2_0)) {
+          EPCISFormat.fromString(EPCIS.XML_FORMAT).get().getMediaType(),
```
Comment (sboeckelmann) [inline src/main/java/io/openepcis/convert/VersionTransformer.java:177]: Use EPCISFormat - EPCIS.XML/JSON_FORMAT may be removed completely

File: src/main/java/io/openepcis/convert/collector/JsonEPCISEventCollector.java [hunk 4]
```diff
@@ -98,11 +103,11 @@ public void start(Map<String, String> context) {
           });
 
       // Start epcisBody object
-      jsonGenerator.writeFieldName("epcisBody");
+      jsonGenerator.writeFieldName(EPCIS.EPCIS_BODY_IN_LOWER_CASE);
```
Comment (sboeckelmann) [inline src/main/java/io/openepcis/convert/collector/JsonEPCISEventCollector.java:106]: This is camelCase not lowercase

File: src/main/java/io/openepcis/convert/collector/JsonEPCISEventCollector.java [hunk 5]
```diff
@@ -118,9 +123,9 @@ public void end() {
       jsonGenerator.writeEndObject(); // End epcisBody
 
       // Write the info related to Context element in JSON
-      jsonGenerator.writeFieldName("@context");
+      jsonGenerator.writeFieldName(EPCIS.CONTEXT);
       jsonGenerator.writeStartArray();
-      jsonGenerator.writeString("https://ref.gs1.org/standards/epcis/2.0.0/epcis-context.jsonld");
+      jsonGenerator.writeString(EPCIS.DEFAULT_CONTEXT);
```
Comment (sboeckelmann) [inline src/main/java/io/openepcis/convert/collector/JsonEPCISEventCollector.java:128]: We might want to move the default jsonld @context from EPCIS.DEFAULT_CONTEXT to EPCISVersion.getDefaultJSONContext

File: src/main/java/io/openepcis/convert/collector/XmlEPCISEventCollector.java [hunk 6]
```diff
@@ -123,9 +126,10 @@ public void end() {
     try {
       // End the EventList, EPCISBody, EPCISDocument and the while document after completing all
       // files writing
-      xmlEventWriter.add(events.createEndElement(new QName("EventList"), null));
-      xmlEventWriter.add(events.createEndElement(new QName("EPCISBody"), null));
-      xmlEventWriter.add(events.createEndElement(new QName("epcis:Document"), null));
+      xmlEventWriter.add(events.createEndElement(new QName(EPCIS.EVENT_LIST_IN_UPPER_CASE), null));
```
Comment (sboeckelmann) [inline src/main/java/io/openepcis/convert/collector/XmlEPCISEventCollector.java:129]: camelCase

File: src/main/java/io/openepcis/convert/collector/XmlEPCISEventCollector.java [hunk 7]
```diff
@@ -123,9 +126,10 @@ public void end() {
...
+      xmlEventWriter.add(events.createEndElement(new QName(EPCIS.EPCIS_BODY_IN_UPPER_CASE), null));
```
Comment (sboeckelmann) [inline src/main/java/io/openepcis/convert/collector/XmlEPCISEventCollector.java:130]: CamelCase

File: src/main/java/io/openepcis/convert/json/JsonToXmlConverter.java [hunk 8]
```diff
@@ -198,11 +200,11 @@ public void convert(
         eventHandler.collectSingleEvent(singleXmlEvent);
       } catch (Exception e) {
         // Loop until the start of the EPCIS EventList array and prepare the XML header elements
-        while (!jsonParser.getText().equals("eventList")) {
+        while (!jsonParser.getText().equals(EPCIS.EVENT_LIST_IN_LOWER_CASE)) {
```
Comment (sboeckelmann) [inline src/main/java/io/openepcis/convert/json/JsonToXmlConverter.java:203]: CamelCase

**Review by sboeckelmann — CHANGES_REQUESTED** [review]

**Comment by Aravinda93** [comment]
@sboeckelmann 

Thanks for the review. Based on your review I have made changes to the project. Please review and approve the PR.

**Review by sboeckelmann — APPROVED** [review]

**Pull request merged**